src/utilities/ApiGatewayAuth.js

```diff
@@ -1,29 +1,42 @@
 import axios from 'axios';
+import qs from 'qs';
 
 export class ApiGatewayAuth {
   constructor(config = {}) {
     this.url = config.url || process.env.API_GATEWAY_AUTH_URL;
     this.clientId = config.clientId || process.env.API_GATEWAY_AUTH_CLIENT_ID;
     this.clientSecret =
       config.clientSecret || process.env.API_GATEWAY_AUTH_CLIENT_SECRET;
-    this.token = config.token;
-    // TODO: Configurable?
-    this.grantType = 'CLIENT_CREDENTIALS';
+    this.token = config.token || false;
+    this.grantType = 'client_credentials';
   }
 
   isAuthenticated = () => !!this.token;
 
   authenticate = async () => {
-    const params = {
+    const body = {
       client_id: this.clientId,
       client_secret: this.clientSecret,
       grant_type: this.grantType,
     };
-    const result = await axios.post(this.url, undefined, {
-      params,
-    });
-    this.token = result.data;
-
-    return this.token;
+    await axios
+      .post(this.url, body)
+      .then((response) => {
+        this.token = response.data;
+        return this.token;
+      })
+      .catch((error) => {
+        console.log('Error:', error.config);
+        if (error.response) {
+          console.log('Error:', error.response.status, error.response.data);
+          Promise.reject(error);
+        }
+        if (error.request) {
+          console.log('Error:', error.request);
+          Promise.reject(error);
+        }
+        console.log('Error:', error.message);
+        Promise.reject(error);
+      });
```

Comment:
```suggestion
      .catch((error) => {
        console.log('Error:', error.config);
        console.log('Error:', error.message);

        if (error.response) {
          console.log('Error:', error.response.status, error.response.data);
        }
        
        if (error.request) {
          console.log('Error:', error.request);
        }
        
        Promise.reject(error);
      });
```

Just a note on style, you don't need to reject multiple times if all you are doing is logging to the console.